### PR TITLE
chore: Remove OAuth env variables from client

### PR DESCRIPTION
**Pull request by sharat87**
Client will get supported OAuth list from the tenant API, instead of from injected env variables like `APPSMITH_OAUTH2_GOOGLE_CLIENT_ID`.

This is a step towards moving OAuth configuration out of env variables completely, and into the backend database, so their configuration can be tenant-wide, instead of instance-wide.

NOTE: Please review/approve only. Please do not merge. It needs an accompanying EE PR to go togather. Thanks!

### CODE REVIEW

File: app/client/src/ce/configs/index.ts [hunk 1]
```diff
@@ -26,7 +24,6 @@ export interface INJECTED_CONFIGS {
     licenseKey: string;
   };
   enableMixpanel: boolean;
-  google: string;
```
Comment (sharat87) [inline app/client/src/ce/configs/index.ts:29]: This is maps API key, got left out, is unused.

File: app/client/src/ce/pages/UserAuth/Login.tsx [hunk 2]
```diff
@@ -85,7 +86,10 @@ export function Login(props: LoginFormProps) {
   const { emailValue: email, error, valid } = props;
   const isFormValid = valid && email && !isEmptyString(email);
   const location = useLocation();
-  const socialLoginList = ThirdPartyLoginRegistry.get();
+  const socialLoginList = [
+    ...useSelector(getThirdPartyAuths),
+    ...ThirdPartyLoginRegistry.get(),
+  ];
```
Comment (sharat87) [inline app/client/src/ce/pages/UserAuth/Login.tsx:92]: Once auth on EE is also moved, we'll only use the selector here.

**Review by ankitakinger — APPROVED** [review]
LGTM

**Review by ankitakinger — APPROVED** [review]
LGTM

**Pull request merged**